acceptance/tests/resource/user/should_purge.rb

```diff
@@ -0,0 +1,40 @@
+test_name "should purge a user"
+confine :except, :platform => /^eos-/ # See ARISTA-37
+confine :except, :platform => /^cisco_/ # See PUP-5828
+tag 'audit:medium',
+    'audit:refactor',  # Use block style `test_run`
+    'audit:acceptance' # Could be done as integration tests, but would
+                       # require changing the system running the test
+                       # in ways that might require special permissions
+                       # or be harmful to the system running the test
+
+name = "pl#{rand(999999).to_i}"
+
+agents.each do |agent|
+  step "ensure the user and group do not exist"
+  agent.user_absent(name)
+  agent.group_absent(name)
+
+  step "create the user"
+  on agent, puppet_resource('user', name, 'ensure=present')
+
+  step "verify the user exists"
+  fail_test "User #{name} was not deleted" unless agent.user_list.include? name
+
+  case agent['platform']
+  when /sles/, /solaris/, /windows/, /osx/, /aix/
+    # no private user groups by default
+  else
+    agent.group_get(name)
+  end
+
+  step "purge the user"
+  on agent, puppet('apply', '-e', %Q|"resources {'user': purge => true, unless_system_user => true }"|)
```

Comment:
I don't think this tests the right thing because it does not test to ensure that Puppet does not delete any users it knows about. Wouldn't you want something like (as an example manifest):
```
user { ['u1', 'u2']:
  ensure => present,
}

resources { 'user':
  purge => true,
  unless_system_user => true
}
```

This would make `u1` and `u2` users that Puppet knows about, so it would remove all users except for `u1` and `u2` + any system users.

It would be good to structure the test to create multiple users that aren't managed by Puppet, some users that are managed by Puppet, and then run the purge and ensure that only the Puppet users + system users remain.